src/vanilla/store.ts

```diff
@@ -331,17 +331,21 @@ const buildStore = (
         }
         return returnAtomValue(aState)
       }
-      // a !== atom
-      const aState = readAtomState(pending, a, dirtyAtoms)
-      if (isSync) {
-        addDependency(pending, atom, atomState, a, aState)
-      } else {
-        const pending = createPending()
-        addDependency(pending, atom, atomState, a, aState)
-        mountDependencies(pending, atom, atomState)
-        flushPending(pending)
+      let aState: AtomState<V>
+      try {
+        // a !== atom
+        aState = readAtomState(pending, a, dirtyAtoms)
```

Comment:
readAtomState shouldn't throw. so, this is okay?
```suggestion
      // a !== atom
      const aState = readAtomState(pending, a, dirtyAtoms)
      try {
```

Reply:
if readAtomState throws, aState can be undefined, which is unexpected, so I think the change ☝️ should be better.